File: leetcode/058-length-of-last-word/solution.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    int lengthOfLastWord(string s) {
        int n = (int)s.size(), count = 0;
        int i = n - 1;
        while (i >= 0 && s[i] == ' ') i--;
        while (i >= 0 && s[i] != ' ') {
            count++;
            i--;
        }
        return count;
    }
};

int main() {
    Solution sol;
    cout << sol.lengthOfLastWord("Hello World") << endl;        // 5
    cout << sol.lengthOfLastWord("   fly me   to   the moon  ") << endl; // 4
    cout << sol.lengthOfLastWord("luffy is still joyboy") << endl; // 6
    return 0;
}

File: leetcode/058-length-of-last-word/solution.py
class Solution:
    def lengthOfLastWord(self, s: str) -> int:
        i = len(s) - 1
        while i >= 0 and s[i] == " ":
            i -= 1
        count = 0
        while i >= 0 and s[i] != " ":
            count += 1
            i -= 1
        return count


if __name__ == "__main__":
    print(Solution().lengthOfLastWord("Hello World"))  # 5
    print(Solution().lengthOfLastWord("   fly me   to   the moon  "))  # 4
    print(Solution().lengthOfLastWord("luffy is still joyboy"))  # 6
